doc/examples/examples.hpp.in

```diff
@@ -0,0 +1,270 @@
+documentation and/or other materials provided with the distribution.
+
+3. Neither the name of the copyright holder nor the names of its
+contributors may be used to endorse or promote products derived from
+this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+******************************<GINKGO LICENSE>*******************************/
+
+
+/**
+ * @page Examples Example programs
+ *
+ * Here you can find Examples programs that demonstrate the usage of Ginkgo.
+ * Some examples are built on another and some are stand-alone and demonstrate a concept of
+ * Ginkgo, which can be used in your own code.
+ *
+ * You can browse the available example programs
+ * <ol>
+ *   <li> as <b><a href="#graph">a graph</a></b> that shows how example programs build upon each other.
+ *   <li> as <b><a href="#list">a list</a></b> that provides a short
+ *     synopsis of each program.
+ *   <li> or <b><a href="#topic">grouped by topic</a></b>.
+ * </ol>
+ *
+ * The easiest way to build the example that you want is to use the script provided:
+ * <code>./build.sh PATH_TO_GINKGO_BUILD_DIR </code>
+ *
+ * By default, Ginkgo is compiled with atleast <code>-DGINKGO_BUILD_REFERENCE=on</code>.
+ * To execute on a GPU, you need to have a GPU on the system and must have
+ * compiled Ginkgo with the <code>-DGINKGO_BUILD_CUDA=ON</code> option.
+ *
+ * Alternatively, you can setup the configuration manually:
+ *
+ * Go to the <code> PATH_TO_GINKGO_BUILD_DIR </code> directory and copy the shared
+ * libraries located in the following subdirectories:
+ * <ol>
+ *   <li> <code>core/</code>
+ *   <li> <code>core/device_hooks/</code>
+ *   <li> <code>reference/</code>
+ *   <li> <code>omp/</code>
+ *   <li> <code>cuda/</code>
+ * </ol>
+ * to this directory.
+ *
+ * Then compile the file with the following command line:
+ * @code{.cpp}
+ * c++ -std=c++11 -o example-* example-*.cpp -I../.. \
+ * -L. -lginkgo -lginkgo_reference -lginkgo_omp -lginkgo_cuda
+ * @endcode
+ * (if ginkgo was built in debug mode, append 'd' to every library name)
+ *
+ * Now you should be able to run the program using:
+ *
+ * @code{.cpp}env LD_LIBRARY_PATH=.:\$\{LD_LIBRARY_PATH\} ./example-*@endcode
```

Comment:
This documentation maybe needs to be updated (example-* form is now gone). Btw, the graph and example page looks excellent with the name replacement, thanks!